File: native-micronaut-api/src/main/java/com/xavierbouclet/nativeapi/ActorController.java
package com.xavierbouclet.nativeapi;

import io.micronaut.http.HttpResponse;
import io.micronaut.http.HttpStatus;
import io.micronaut.http.annotation.*;
import java.util.*;

@Controller(value = "/actors")
record ActorController(ActorService service) {

    @Get
    public HttpResponse<Iterable<Actor>> list()  {
        return HttpResponse.status(HttpStatus.OK).body(service.list());
    }

    @Get("/{id}")
     public HttpResponse<Actor> get(@PathVariable UUID id ) {
        return HttpResponse.status(HttpStatus.OK).body(service.get(id));
    }

    @Post
     public HttpResponse<Actor> add(@Body Actor actor ) {
        return HttpResponse.status(HttpStatus.CREATED).body(service.add(actor));
    }

    @Put
     public HttpResponse<Actor> modify(@Body Actor actor) {
        return HttpResponse.status(HttpStatus.ACCEPTED).body(service.modify(actor));
    }

    @Delete("/{id}")
     public HttpResponse<UUID> delete(@PathVariable UUID id ) {
        return HttpResponse.status(HttpStatus.OK).body(service.delete(id));
    }

    @Put("/{id}/movies/{movieId}")
     public HttpResponse<Actor> addActorToMovie(@PathVariable UUID id , @PathVariable UUID movieId) {
        return HttpResponse.status(HttpStatus.ACCEPTED).body(service.addMovieToActor(id, movieId));
    }

    @Delete("/{id}/movies/{movieId}")
     public HttpResponse<Actor> deleteActorFromMovie(@PathVariable UUID id , @PathVariable UUID movieId){
        return HttpResponse.status(HttpStatus.OK).body(service.removeMovieFromActor(id, movieId));
    }

}

File: native-micronaut-api/src/main/kotlin/com/xavierbouclet/nativeapi/ActorController.kt
package com.xavierbouclet.nativeapi

import io.micronaut.http.HttpResponse
import io.micronaut.http.HttpStatus
import io.micronaut.http.annotation.*
import java.util.*

@Controller(value = "/actors")
class ActorController(val service: ActorService) {

    @Get
    fun list(): HttpResponse<*> {
        return HttpResponse.status<Any>(HttpStatus.OK).body(service.list())
    }

    @Get("/{id}")
    fun list(@PathVariable id: UUID): HttpResponse<*> {
        return HttpResponse.status<Any>(HttpStatus.OK).body(service.get(id))
    }

    @Post
    fun add(@Body actor: Actor): HttpResponse<*>? {
        return HttpResponse.status<Any>(HttpStatus.CREATED).body(service.add(actor))
    }

    @Put
    fun modify(actor: Actor): HttpResponse<*>? {
        return HttpResponse.status<Any>(HttpStatus.ACCEPTED).body(service.modify(actor))
    }

    @Delete("/{id}")
    fun delete(@PathVariable id: UUID): HttpResponse<*>? {
        return HttpResponse.status<Any>(HttpStatus.OK).body(service.delete(id))
    }

    @Put("/{id}/movies/{movieId}")
    fun addActorToMovie(@PathVariable id: UUID, @PathVariable movieId: UUID): HttpResponse<*>? {
        return HttpResponse.status<Any>(HttpStatus.ACCEPTED).body(service.addMovieToActor(id, movieId))
    }

    @Delete("/{id}/movies/{movieId}")
    fun deleteActorFromMovie(@PathVariable id: UUID, @PathVariable movieId: UUID): HttpResponse<*>? {
        return HttpResponse.status<Any>(HttpStatus.OK).body(service.removeMovieFromActor(id, movieId))
    }

}
